migrate/src/main/scala/migrate/utils/ScalafixService.scala

```diff
@@ -0,0 +1,126 @@
+package migrate.utils
+
+import scala.jdk.CollectionConverters._
+import scala.util.Try
+
+import buildinfo.BuildInfo
+import coursier._
+import migrate.AbsolutePath
+import migrate.Classpath
+import migrate.utils.ScalaExtensions.OptionalExtension
+import scalafix.interfaces.Scalafix
+import scalafix.interfaces.ScalafixEvaluation
+
+final case class ScalafixService(
+  scalafix: Scalafix,
+  sources: Seq[AbsolutePath],
+  compilerOptions: Seq[String],
+  classpath: Classpath,
+  targetRootSemantic: AbsolutePath,
+  toolClasspath: Classpath
+) {
+  import ScalafixService._
+  lazy val scalafixClassLoader: ClassLoader = scalafix.getClass().getClassLoader()
+
+  def inferExplicitAndImplicits(): Try[ScalafixEvaluation] = evaluate(addExplicitResultTypesAndImplicits)
```

Comment:
This name is really confusing... I suggest `inferTypesAndImplicits`.

Reply:
ok